Request for add-tech branch merge to master

DannyNicholas opened the pull request:
Please review the pull request changes for the add-tech branch. This is ready to merge with the master if approved.

Commit:
adding new tech component.

Commit:
Fixing merge issues

Commit:
stop duplicate tech being added.

Commit:
modified add tech form to include form state

Commit:
fixed merge issues

Commit:
fixed add new tech bug after merge.

Commit:
add new technology tests

Commit:
lint errors fixed

Commit:
add UI warning if duplicate tech entered

mtinning commented on app/components/TechPoll.js:
```diff
@@ -1,10 +1,11 @@
 import * as actionCreators from '../store/action-creators'
 
-const TechPoll = ({ tech, submitVote }) => {
+const TechPoll = ({ tech, submitVote, addNewTech }) => {
```
This passing all of the props down from the parent is getting tedious. We should refactor these a bit so that the individual components connect to the stores (turn the components into 'containers'). We'll work together on this, no need to change it for this request

mtinning commented on app/components/AddNewTech.js:
```diff
@@ -0,0 +1,71 @@
+              onChange={handleTechChange}
+              required
+            />
+            <font color="red"><strong>{this.state.warning}</strong></font>
```
Rather than using `<font />` and `<strong/>` elements, we should use `<span class="validation-error" />` and use css to style. You'll need to add a `add-new-tech.scss` file alongside, and include that in the `index.scss` file.

In addition, we should not be rendering this element at all if there is no warning - you can do this with something like:

``` javascript
{ !!this.state.warning ? <span>{this.state.warning}</span> : null }
```

Not sure about the nested braces though, works but looks pretty horrendous!

mtinning commented on app/store/reducer.js:
```diff
@@ -8,10 +8,24 @@ function submitVote(item, vote) {
+  const index = state.get('tech').findIndex(t => t.get('name') === item.name)
+
+  if (index >= 0) {
+    // console.log('Tech: \''+ item.name + '\' already exists')
```
Always like to remove commented-out code

mtinning commented:
Made a few comments @DannyNicholas, take a look

mtinning commented:
Also, you'll see that the branch has conflicts - you'll need to merge with master before this can be merged

Commit:
merge with master and fix conflicts.

Commit:
improved form validation with CSS

DannyNicholas commented:
@mtinning - merged and fixed conflicts. Form now uses new categories correctly and improved form validation with CSS. Also removed commented out code. Please take a look when you have a sec.  Thanks!

mtinning commented:
Looks good! :ship:
